zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs

```diff
@@ -0,0 +1,493 @@
+
+#[derive(Clone, Debug)]
+pub(crate) struct ErrorOOGCallGadget<F> {
+    opcode: Cell<F>,
+    tx_id: Cell<F>,
+    is_static: Cell<F>,
+    gas: Word<F>,
+    callee_address: Word<F>,
+    value: Word<F>,
+    is_warm: Cell<F>,
+    is_warm_prev: Cell<F>,
+    value_is_zero: IsZeroGadget<F>,
+    cd_address: MemoryAddressGadget<F>,
+    rd_address: MemoryAddressGadget<F>,
+    memory_expansion: MemoryExpansionGadget<F, 2, N_BYTES_MEMORY_WORD_SIZE>,
+    balance: Word<F>,
+    callee_nonce: Cell<F>,
+    callee_code_hash: Cell<F>,
+    is_empty_nonce_and_balance: BatchedIsZeroGadget<F, 2>,
+    is_empty_code_hash: IsEqualGadget<F>,
+    insufficient_gas: LtGadget<F, N_BYTES_GAS>,
+    restore_context: RestoreContextGadget<F>,
+}
+
+impl<F: Field> ExecutionGadget<F> for ErrorOOGCallGadget<F> {
+    const NAME: &'static str = "ErrorOutOfGasCall";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::ErrorOutOfGasCALL;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let opcode = cb.query_cell();
+        cb.opcode_lookup(opcode.expr(), 1.expr());
+        // TODO: add CallCode etc. when handle ErrorOutOfGasCALLCODE in furture
+        // implementation
+        cb.require_equal(
+            "ErrorOutOfGasCall opcode is Call",
+            opcode.expr(),
+            OpcodeId::CALL.expr(),
+        );
+
+        let gas_word = cb.query_word();
+        let callee_address_word = cb.query_word();
+        let value = cb.query_word();
+        let cd_offset = cb.query_cell();
+        let cd_length = cb.query_rlc();
+        let rd_offset = cb.query_cell();
+        let rd_length = cb.query_rlc();
+        let balance = cb.query_word();
+
+        let tx_id = cb.call_context(None, CallContextFieldTag::TxId);
+
+        let is_static = cb.call_context(None, CallContextFieldTag::IsStatic);
+
+        // Lookup values from stack
+        cb.stack_pop(gas_word.expr());
+        cb.stack_pop(callee_address_word.expr());
+        cb.stack_pop(value.expr());
+        cb.stack_pop(cd_offset.expr());
+        cb.stack_pop(cd_length.expr());
+        cb.stack_pop(rd_offset.expr());
+        cb.stack_pop(rd_length.expr());
+        cb.stack_push(0.expr());
+
+        // Recomposition of random linear combination to integer
+        let callee_address =
+            from_bytes::expr(&callee_address_word.cells[..N_BYTES_ACCOUNT_ADDRESS]);
+        let cd_address = MemoryAddressGadget::construct(cb, cd_offset, cd_length);
+        let rd_address = MemoryAddressGadget::construct(cb, rd_offset, rd_length);
+        let memory_expansion =
+            MemoryExpansionGadget::construct(cb, [cd_address.address(), rd_address.address()]);
+
+        // Add callee to access list
+        let is_warm = cb.query_bool();
+        let is_warm_prev = cb.query_bool();
+        cb.account_access_list_read(
+            tx_id.expr(),
+            callee_address.clone(),
+            is_warm.expr(),
+            is_warm_prev.expr(),
+        );
```

Comment:
Then we can get rid of `is_warm_prev`

```suggestion
        let is_warm = cb.query_bool();
        cb.account_access_list_read(
            tx_id.expr(),
            callee_address.clone(),
            is_warm.expr(),
        );
```